src/client/cmd/info.cpp

```diff
@@ -149,22 +36,20 @@ mp::ReturnCode cmd::Info::run(mp::ArgParser* parser)
     }
 
     auto on_success = [this](mp::InfoReply& reply) {
+        std::string output;
+
         if (format_type == "json")
         {
-            cout << toJson(reply).toStdString();
+            JsonOutput json_output;
+            output = json_output.process_info(reply);
```

Comment:
Now that there's a proper interface, multipass::Client should have a member variable map that holds all the different Formatters.

`std::map<std::string, std::unique_ptr<mp::Formatter>> formatters;`

`std::map` because we want ordered entries so that the "format" option help can be dynamically generated from this table, where the first entry is the default.

The the mp::Client constructor can instantiate it as:

```
{{"table", std::make_unique<TableFormatter>,
{"json", std::make_unique<JsonFormatter>}}
```

Then Command can hold a reference to the map.
A utility method in Command can validate the format option from the map, which can be called by the concrete command at parsing time, something like:

```
Formatter* Command::formatter_for(std::string format)
{
   auto entry = formatters.find(format);
   if (entry != formatters.end())
       return entry.second;
   return nullptr;
}
```
In cmd::Info::parse_args(...) then:

```
chosen_formatter = formatter_for(format);
if (chosen_formatter == nullptr)
{
    cerr << "Invalid format type given.\n";
    return ParseCode::CommandLineError;
}
```

Finally, this section would just simply do:

`cout << chosen_formatter.format(reply);`